src/js/setup-text-tracks.js

```diff
@@ -0,0 +1,136 @@
+import dashjs from 'dashjs';
+
+/*
+ * Attach text tracks from dash.js to videojs
+ *
+ * @param {videojs} player the videojs player instance
+ * @param {array} tracks the tracks loaded by dash.js to attach to videojs
+ *
+ * @private
+ */
+function attachDashTextTracksToVideojs(player, tracks) {
+  const mediaPlayer = player.dash.mediaPlayer;
+
+  const trackDictionaryKeyedByTextTrack = new WeakMap();
```

Comment:
I don't think it's a good idea to use WeakMap yet, still. To support the widest field of browsers possible, we can't use WeakMap and WeakMap isn't really polyfillable.